src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php

```diff
@@ -164,11 +194,17 @@ private function initializeListeners(string $eventName)
     private function initializeSubscribers()
     {
         $this->initializedSubscribers = true;
-        foreach ($this->subscribers as $id => $subscriber) {
+        foreach ($this->subscribers as $subscriber) {
+            if (\is_array($subscriber)) {
+                $this->addEventListener(...$subscriber);
```

Comment:
What about turning this into `array{string[], \Closure}`, where the Closure would return the listener? That would allow making listeners lazy. Could that have some benefits, eg to help build graph of circular deps?

Reply:
if I read the pass correctly, this is already lazy, because we only pass the id here?
then it means we miss a call to `$this->container->get()` (and related tests)?

Reply:
`addEventListener` accept either a string or an object (sorry my constructor doc block was not exhaustive).
This behavior is already handled and tested in `initializeListeners` https://github.com/symfony/symfony/blob/5c53f99f4cb5643e1720df3a1923d476d1b94265/src/Symfony/Bridge/Doctrine/ContainerAwareEventManager.php#L182-L192

Here I just dequeue the subscribers stack in order to resolve listeners priority, but Keep the previous business logic.